cpp/Esri/ArcGISRuntime/Toolkit/GenericListModel.h

```diff
@@ -0,0 +1,343 @@
+/*******************************************************************************
+ *  Copyright 2012-2020 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_GENERICLISTMODEL_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_GENERICLISTMODEL_H
+
+#include <QAbstractListModel>
+#include <QMetaObject>
+
+#include <type_traits>
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+namespace Toolkit
+{
+
+/*!
+ * \internal
```

Comment:
This is a tremendous amount of documentation for an internal class.

Reply:
It sure is. In our current workflow it's all part of the user source code and accessible/usable to them. 

While we can inform the user that something is an implementation detail, we can't prevent them from using/inspecting it. I have gone for a blanket approach of documenting everything as if it were fair game to the consumer.

Reply:
@anmacdonald that's fair, but for the classes that are purely implementation details, I would make it obvious other than just `\internal` that these are internal classes and nobody need understand how/why they work or exist.

This level of documentation would grab my eye as something that is paramount that I would need to know about and understand as a user.